### PR TITLE
Code Style Cleanup

### PR DESCRIPTION
**Pull request by davkean**
This includes https://github.com/dotnet/roslyn-project-system/pull/1002, so any rule set changes/target changes can be skipped over.

This is a mop-up of the last of the remaining violations of the code style rules. The rule set will be turned on after this has been merged.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/GlobalSuppressions.cs [hunk 1]
```diff
@@ -10,3 +10,5 @@
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0018:Inline variable declaration", Justification = "<Pending>", Scope = "member", Target = "~M:Microsoft.VisualStudio.Shell.Interop.IVsHierarchyFactory.ImplementGetProperty(Microsoft.VisualStudio.Shell.Interop.IVsHierarchy,Microsoft.VisualStudio.Shell.VsHierarchyPropID,System.Int32)")]
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0018:Inline variable declaration", Justification = "<Pending>", Scope = "member", Target = "~M:Microsoft.VisualStudio.Shell.Interop.IVsHierarchyFactory.ImplementGetProperty(System.Int32)~Microsoft.VisualStudio.Shell.Interop.IVsHierarchy")]
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0018:Inline variable declaration", Justification = "<Pending>", Scope = "member", Target = "~M:Microsoft.VisualStudio.Shell.Interop.IVsProjectFactory.ImplementOpenItemWithSpecific(Microsoft.VisualStudio.Shell.Interop.IVsProject4,System.Guid,System.Guid,System.Int32)")]
+
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/GlobalSuppressions.cs]: @mavasani Is this by design this whitespace was inserted?

Comment (mavasani) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/GlobalSuppressions.cs]: I think the fixer tries to scan existing file and adds an extra new line only for first attribute - so seems like a minor bug to me. If you have a self contained repro, can you file a bug?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesSubTreeProviderBase.cs [hunk 2]
```diff
@@ -364,11 +364,10 @@ public virtual Task<IEnumerable<IDependencyNode>> SearchAsync(IDependencyNode no
                 {
                     string unresolvedItemSpec = resolvedReferenceRuleChanges.Before
                                                     .Items[removedItemSpec][OriginalItemSpecPropertyName];
-                    IProjectRuleSnapshot unresolvedReferenceSnapshot = null;
                     string unresolvedItemType = GetUnresolvedReferenceItemType(unresolvedItemSpec,
-                                                                            updatedUnresolvedSnapshots,
-                                                                            catalogs,
-                                                                            out unresolvedReferenceSnapshot);
+                                                        updatedUnresolvedSnapshots,
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesSubTreeProviderBase.cs]: Filed: https://github.com/dotnet/roslyn/issues/16198

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesSubTreeProviderBase.cs [hunk 3]
```diff
@@ -398,11 +397,10 @@ public virtual Task<IEnumerable<IDependencyNode>> SearchAsync(IDependencyNode no
                     }
 
                     var originalItemSpec = properties[OriginalItemSpecPropertyName];
-                    IProjectRuleSnapshot unresolvedReferenceSnapshot = null;
                     var itemType = GetUnresolvedReferenceItemType(originalItemSpec,
-                                                                  updatedUnresolvedSnapshots,
-                                                                  catalogs,
-                                                                  out unresolvedReferenceSnapshot);
+                                              updatedUnresolvedSnapshots,
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesSubTreeProviderBase.cs]: Filed: https://github.com/dotnet/roslyn/issues/16198

File: build/Rulesets/NonShippingProjectRules.ruleset [hunk 4]
```diff
@@ -15,4 +15,9 @@
     <Rule Id="RS0006" Action="None" />
     <Rule Id="RS0023" Action="None" />
   </Rules>
+   <Rules AnalyzerId="Microsoft.CodeAnalysis.CSharp.Features" RuleNamespace="Microsoft.CodeAnalysis.CSharp.Features">
+     <!-- Naming styles is too noisy as it fires on all async tests -->
```
Comment (333fred) [inline build/Rulesets/NonShippingProjectRules.ruleset]: Is there any way we can turn it off for only test projects?

Comment (davkean) [inline build/Rulesets/NonShippingProjectRules.ruleset]: NonShippingProjectRules is effectively that, it applies to all non-shipping projects, which in our tree is the majority of the test projects.

Comment (333fred) [inline build/Rulesets/NonShippingProjectRules.ruleset]: Ah, got it. Thanks.

**Pull request merged**